tree/treeplayer/src/TTreeIndex.cxx

```diff
@@ -638,3 +638,50 @@ void TTreeIndex::SetTree(TTree *T)
    fTree = T;
 }
 
+/// \cond HIDDEN_SYMBOLS
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Create a deep copy of the TTreeIndex
+/// The new index is not attached to any tree. It is the responsibility of the
+/// caller to attach it to a tree if necessary.
+std::unique_ptr<TVirtualIndex> TTreeIndex::MakeIndexCopy() const
+{
+   auto index = std::make_unique<TTreeIndex>();
```

Comment:
I specifically do not think that this is a good idea. And I also wrote it in the documentation of the function. The newly created index should not be attached to the same TTree as the one that copied it, because that TTree itself will have a reference to the previous index, so now two indexes are pointing to the same tree, I believe that is recipe for trouble

Reply:
That is a salient point.  It is also relates to how the `TTreeFormula` are being set.  Let me think.